core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -374,6 +374,7 @@ class SparkContext(config: SparkConf) extends Logging {
   private[spark] def cleaner: Option[ContextCleaner] = _cleaner
 
   private[spark] var checkpointDir: Option[String] = None
+  config.getOption(CHECKPOINT_DIR.key).foreach(setCheckpointDir)
```

Comment:
We should do this only after `_hadoopConfiguration` has been initialized.
Move it to after line [535 here](https://github.com/apache/spark/blob/f40ffe0f97a2e753878af8f8f47038272a1e1657/core/src/main/scala/org/apache/spark/SparkContext.scala#L535) ?